packages/strapi/lib/middlewares/parser/index.js

```diff
@@ -2,21 +2,23 @@
 
 const body = require('koa-body');
 const qs = require('qs');
+const { omit } = require('lodash');
 
 /**
  * Body parser hook
  */
-const addQsParser = app => {
+const addQsParser = (app, settings) => {
   Object.defineProperty(app.request, 'query', {
     configurable: false,
     enumerable: true,
     /*
      * Get parsed query-string.
      */
     get() {
+      const { arrayLimit } = settings;
       const qstr = this.querystring;
       const cache = (this._querycache = this._querycache || {});
-      return cache[qstr] || (cache[qstr] = qs.parse(qstr, { depth: 20 }));
+      return cache[qstr] || (cache[qstr] = qs.parse(qstr, { arrayLimit, depth: 20 }));
```

Comment:
can you move the depth to the default to and pass the entire setting object so a user can use extra params